server/routes.js

```diff
@@ -50,67 +88,21 @@ module.exports = [{
-      log.info('auth/complete invoked');
-      // HUGE TODO verify the session cookie matches the 'state' nonce in the query
-      var tokenPayload = {
-        client_id: config.get('server.oauth.clientId'),
-        client_secret: config.get('server.oauth.clientSecret'),
-        code: request.query.code
+      // Bell has obtained the oauth token, used it to get the profile, and
+      // that profile is available as request.auth.credentials.profile.
+      // Now, use it to set a cookie.
+      log.verbose('inside auth/complete');
+      log.verbose('request.auth.credentials is ' + JSON.stringify(request.auth.credentials));
+
+      var session = {
+        fxaId: request.auth.credentials.profile.fxaId
       };
-      // 1. swap code for token
-      wreck.post(config.get('server.oauth.tokenEndpoint'),
-        { payload: JSON.stringify(tokenPayload) },
-        function(err, res, payload) {
-          if (err) {
-            log.info('token server error: ' + err);
-            // TODO something went wrong, try again? throw AppError?
-            return reply.redirect('/');
-          }
-          if (!payload) {
-            log.info('token server returned empty response');
-            return reply.redirect('/');
-          }
-          // TODO can Joi ensure JSON.parse doesn't throw? #43
-          var pay = JSON.parse(payload);
-          var accessToken = pay && pay['access_token'];
-          log.debug('token server response: ' + payload);
-          log.debug('token server response http code: ' + res.statusCode);
-          if (!accessToken) {
-            log.info('no access token found in token server response');
-            return reply.redirect('/');
-          }
-          // 2. use the token to obtain profile data
-          wreck.get(config.get('server.oauth.profileEndpoint'),
-            { headers: {'authorization': 'Bearer ' + accessToken}},
-            function(err, res, payload) {
-              if (err) {
-                log.info('profile server error: ' + err);
-                return reply.redirect('/');
-              }
-              if (!payload) {
-                log.info('profile server returned empty response');
-                return reply.redirect('/');
-              }
-              log.info('profile server response: ' + payload);
-              // TODO can Joi ensure JSON.parse doesn't throw? #43
-              var pay = JSON.parse(payload);
-              db.createUser(pay.uid, pay.email, accessToken, function(err) {
-                if (err) {
-                  log.info('user creation failed: ' + err);
-                  return reply.redirect('/');
-                }
-                request.auth.session.set({fxaId: payload.uid});
-                reply.redirect('/');
-              });
-            }
-          );
-        }
-      );
+      var duration = config.get('server.session.duration');
+      if (duration > 0) {
+        session.expiresAt = new Date(new Date().getTime() + duration);
```

Comment:
I'm not sure I completely understand what's happening here. Maybe a few comments would help?